processor/src/main/java/org/mapstruct/ap/internal/processor/Jsr330ComponentProcessor.java

```diff
@@ -58,21 +60,35 @@ protected boolean requiresGenerationOfDecoratorClass() {
     }
 
     private Annotation singleton() {
-        return new Annotation( getTypeFactory().getType( "javax.inject.Singleton" ) );
+        return new Annotation( getType( "Singleton" ) );
     }
 
     private Annotation named() {
-        return new Annotation( getTypeFactory().getType( "javax.inject.Named" ) );
+        return new Annotation( getType( "Named" ) );
     }
 
     private Annotation namedDelegate(Mapper mapper) {
         return new Annotation(
-            getTypeFactory().getType( "javax.inject.Named" ),
+            getType( "Named" ),
             Collections.singletonList( '"' + mapper.getPackageName() + "." + mapper.getName() + '"' )
         );
     }
 
     private Annotation inject() {
-        return new Annotation( getTypeFactory().getType( "javax.inject.Inject" ) );
+        return new Annotation( getType( "Inject" ) );
+    }
+
+    private Type getType(String simpleName) {
```

Comment:
I noticed that this implementation differs from the `JakartaComponentProcessor`. Perhaps factor this method out as static (util) to be used in both the `JSR330ComponentProcessor` and this one.

Reply:
It differs because my goal was for the `JSR330ComponentProcessor` to support both javax and jakarta, with a precedence over javax. This means that if you do not have `javax` available and you use `jsr330` MapStruct will create the correct mappers with Jakarta. This would mean that you don't need to do any changes in your mappers if you just switch dependencies.

However, if you explicitly use the `jakarta` component model then we are only going to use jakarta annotations, won't do any special fallbacks.